test/correctness/simd_op_check.h

```diff
@@ -0,0 +1,301 @@
+#ifndef SIMD_OP_CHECK_H
+#define SIMD_OP_CHECK_H
+
+#include "Halide.h"
+#include "test/common/halide_test_dirs.h"
+#include "sys/time.h"
+#include <fstream>
+
+using namespace Halide;
+using namespace Halide::ConciseCasts;
+
+//namespace {
+
+// This tests that we can correctly generate all the simd ops
+using std::vector;
+using std::string;
+
+
+const Var x{"x"}, y{"y"};
+
+struct TestResult {
+    string op;
+    string error_msg;
+};
+
+struct Task {
+    string op;
+    string name;
+    int vector_width;
+    Expr expr;
+};
+
+size_t num_threads = Halide::Internal::ThreadPool<void>::num_processors_online();
```

Comment:
Declaring a global in a .h file in this manner is ~never OK

Reply:
Fixed.